parsl/executors/taskvine/manager.py

```diff
@@ -364,18 +373,26 @@ def _taskvine_submit_wait(ready_task_queue=None,
                         if spec.parsl_name in parsl_file_name_to_vine_file:
                             task_in_file = parsl_file_name_to_vine_file[spec.parsl_name]
                         else:
-                            task_in_file = m.declare_file(spec.parsl_name, cache=spec.cache, peer_transfer=True)
+                            task_in_file = _handle_file_declaration_protocol(m, spec.parsl_name, spec.cache)
                             parsl_file_name_to_vine_file[spec.parsl_name] = task_in_file
-                        t.add_input(task_in_file, spec.parsl_name)
+                        logger.debug("Adding input file {}, {} to TaskVine".format(spec.parsl_name, task.executor_id))
+                        if spec.netloc == '':
+                            t.add_input(task_in_file, spec.parsl_name)
+                        else:
+                            t.add_input(task_in_file, spec.netloc)
```

Comment:
I'm a bit unclear what the use of spec.netloc is here... is it the netloc from the user specified URL? does that mean that if you're saying http://foo/bar.txt, the task-vine side filename will be foo?  (rather than bar.txt or something else?)

Reply:
This change, and other netloc vs. filename related items are a result of some tricky naming items going on here.

My observation was that the current url parsing will fill in parsl_file.netloc with the "remote name", so bar.txt in your example. This is what we wish to refer to it as within the application